README.md

```diff
@@ -1,106 +1,193 @@
+3. create and update a key-value pair.
+
+The API is designed to hide data-plane differences between different key-value stores. The control-plane behavior/functionality (e.g., including cluster management, data consistency, replication, and sharding) are not specified, and are provider specific.
+
+### Goals
 
-### Goals [or Motivating Use Cases, or Scenarios]
+The primary goal of this API is to allow users to use WASI programs to access key-value stores. It abstracts away the complexity of building stateful services that need to access key-value stores. In other words, it decouples the application code from the specific key-value store APIs. This allows WASI programs to be portable across different key-value store that supports this interface, be it on-prem, in the cloud, or in edge devices.
 
-[What is the end-user need which this project aims to address?]
+The second goal of this API is to abstract away the network stack, allowing applications that need to access key-value stores to not worry about what network protocol is used to access the key-value store.
 
 ### Non-goals
 
-[If there are "adjacent" goals which may appear to be in scope but aren't, enumerate them here. This section may be fleshed out as your design progresses and you encounter necessary technical and other trade-offs.]
+- Cover all application requirements for a key-value store. Instead, it focuses on the most common use cases. It allows providers to extend the API to support more use cases.
+- Transactional semantics.
+- Data consistency.
+- Data replication.
+- Data sharding.
+- Cluster management.
+- Monitoring
 
 ### API walk-through
 
-[Walk through of how someone would use this API.]
+#### Use case 1
 
-#### [Use case 1]
+Imagine you have an HTTP handler that needs to persist some data to a key-value store. The handler needs to be able to retrieve, delete, and update the data in the key-value store. The following Rust code shows how you can use the WASI Key-Value Store API to in the handler.
 
-[Provide example code snippets and diagrams explaining how the API would be used to solve the given problem]
+```rust
+let value = b"value1";
+kv1::create_or_update("key", value)?;
+let val = kv1::get("key")?;
+assert_eq!(val, value);
 
-#### [Use case 2]
-
-[etc.]
+kv1::delete("key")?;
+```
 
 ### Detailed design discussion
 
-[This section should mostly refer to the .wit.md file that specifies the API. This section is for any discussion of the choices made in the API which don't make sense to document in the spec file itself.]
+```go
+interface "wasi:kv/data/crud" {
+  use { Error, payload, key } from "wasi:kv/types"
+
+  get: func(key: key) -> result<payload, Error>
+
+  create-or-update: func(key: key, value: stream<u8>) -> result<_, Error>
 
-#### [Tricky design choice #1]
+  delete: func(key: key) -> result<_, Error>
+}
 
-[Talk through the tradeoffs in coming to the specific design point you want to make.]
+interface "wasi:kv/data/increment" {
```

Comment:
I think the main reason why this is a separate interface is for atomic operations. The slide deck presentation shows the `wasi:kv/data/atomic` interface. I'd rather see the atomic interface with an operation called `increment` (we don't need the short `incr` , names are cheap).

Reply:
yeah the PR is out of date. I will create the increment name to atomic to sync with the slides.